pilot/pkg/model/config.go

```diff
@@ -267,12 +267,12 @@ type IstioConfigStore interface {
 	QuotaSpecByDestination(instance *ServiceInstance) []Config
 
 	// AuthenticationPolicyByDestination selects authentication policy associated
-	// with a service + port.
+	// with a service (or labels if specified) + port.
 	// If there are more than one policies at different scopes (global, namespace, service)
 	// the one with the most specific scope will be selected. If there are more than
 	// one with the same scope, the first one seen will be used (later, we should
 	// have validation at submitting time to prevent this scenario from happening)
-	AuthenticationPolicyByDestination(service *Service, port *Port) *Config
+	AuthenticationPolicyByDestination(service *Service, port *Port, labels Labels) *Config
```

Comment:
[nit] We may want to change this name. 'ByDestination' was wrongly named (my bad) and may cause confusion, especially with label selector now. `AuthenticationPolicyForWorkload` may be a better choice.

Reply:
Done